### PR TITLE
gitlab4j#1023 Add Field Iteration to Issue.

### PR DESCRIPTION
**Pull request by KoCoder**
+ Iteration Class
+ Getter and Setter for Iteration field

closes https://github.com/gitlab4j/gitlab4j-api/issues/1023

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/models/Iteration.java
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/Iteration.java:1]: There should be a test case with an example JSON

* new json example in `src/test/resources/org/gitlab4j/api/`. Example `iteration.json`
* new test method `TestGitLabApiBeans`.

File: src/main/java/org/gitlab4j/api/models/AbstractIssue.java
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/AbstractIssue.java:1]: To verify:

This `AbstractIssue` class is shared between multiple concrete classes.

we should verify where the `iteration` attribute is available.

**Review by jmini — CHANGES_REQUESTED** [review]
The change looks good to me. Just a test case is missing.

File: src/main/java/org/gitlab4j/api/models/AbstractIssue.java
Comment (jmini) [reply src/main/java/org/gitlab4j/api/models/AbstractIssue.java:1]: In `EpicIssue` it is available.

```
GET <server>/api/v4/groups/:id/epics/:epic_iid/issues
```

In `LinkedIssue` it is available as well:
```
GET <server>/api/v4/projects/:project/issues/:issue_iid/links
```

So adding it in `AbstractIssue` is correct

File: src/main/java/org/gitlab4j/api/models/Iteration.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        }
+	}
+
+	private Long id;
+	private Long iid;
+	private Long sequence;
+	private Long groupId;
+	private String title;
+	private String descripion;
+	private IterationState state;
```
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/Iteration.java:38]: In the json, the state is an integer:

https://gitlab.com/gitlab-org/gitlab/-/blob/4db027c5a18a8638bb0e6103b49010975058a9f8/doc/api/iterations.md#L56

File: src/main/java/org/gitlab4j/api/models/Iteration.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            return (enumHelper.toString(this));
+        }
+	}
+
+	private Long id;
+	private Long iid;
+	private Long sequence;
+	private Long groupId;
+	private String title;
+	private String descripion;
```
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/Iteration.java:37]: There is a typo: Should be `description`

File: src/main/java/org/gitlab4j/api/models/Iteration.java (hunk 1 above)
Comment (jmini) [reply src/main/java/org/gitlab4j/api/models/Iteration.java:38]: From the code:
https://gitlab.com/gitlab-org/gitlab/-/blob/4db027c5a18a8638bb0e6103b49010975058a9f8/ee/app/models/iteration.rb#L14-18

It seems that there are only 3 possible values.

**Comment by jmini** [comment]
@KoCoder you did not grant `Maintainers are allowed to edit this pull request` on this PR.

I made the requested changes in https://github.com/jmini/gitlab4j-api/commit/f3cb01cd39a435cd506bf3120f7f03777267fcd5

Can you add that to this PR (on the branch `heek-feature-gitlab4j-1023-iterations`)?

Or should I re-create an other PR to get this change integrated?

**Comment by KoCoder** [comment]
Hey, I've just cherry-picked your changes onto our feature branch.

**Review by jmini — APPROVED** [review]

**Comment by PaulWoitaschek** [comment]
@jmini 
Could you cut a release with this change included? Alternatively: Do snapshots get published of gitlab4j?

**Pull request merged**